From 8ca1128e52961e82afcb510800ceeaec0ade731c Mon Sep 17 00:00:00 2001
From: Peter Harliman Liem <peter.harliman.liem@intel.com>
Date: Mon, 9 Sep 2019 14:39:50 +0800
Subject: [PATCH] DRVLIB_SW-2884 - Fix klocwork issue in phy-wan-xpcs

- debugfs_create_dir() may return error or null
- change mode to 0600
---
 drivers/phy/intel/phy-intel-wan-xpcs.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/phy/intel/phy-intel-wan-xpcs.c b/drivers/phy/intel/phy-intel-wan-xpcs.c
index bd9b25e591c3..40b99239c466 100644
--- a/drivers/phy/intel/phy-intel-wan-xpcs.c
+++ b/drivers/phy/intel/phy-intel-wan-xpcs.c
@@ -176,11 +176,10 @@ static int intel_wan_xpcs_phy_debugfs_init(struct intel_wan_xpcs_phy *priv)
 
 	strlcpy(wan_xpcs_dir, priv->dev->of_node->name, sizeof(wan_xpcs_dir));
 	priv->debugfs = debugfs_create_dir(wan_xpcs_dir, NULL);
-
-	if (!priv->debugfs)
+	if (IS_ERR_OR_NULL(priv->debugfs))
 		return -ENOMEM;
 
-	if (!debugfs_create_file("wan_xpcs_serdes", 0644, priv->debugfs,
+	if (!debugfs_create_file("wan_xpcs_serdes", 0600, priv->debugfs,
 				 priv, &intel_wan_xpcs_fops))
 		goto __debugfs_err;
 
